PocketTutor-Team/pockettutor-app
# Feature/tutor response screen

# Implement the TutorResponseLesson screen so that tutor can accept a pending lesson

### PR Description
This PR add the screen TutorResponseLesson (see an example of it UI at the end of the description). It also implement navigation to this screen from the RequestedLesson screen.
Moreover it also update the LessonState with the state needed to complete the lifecycle of a lesson. You can have a look at the Lesson State diagram that is in the wiki to see how we go from one state to another.

### Modified files
- MainActivity.kt => update the navigation to be able to navigate to the TutorResponseLesson screen
- LessonStatus.kt => update the lesson status with the separation of the requested state when it is a student who made the request and when it is a tutor
- LessonViewModel.kt => fix a merge problem with the main (doesn't make any new updates)
- ListProfilesViewModel.kt => add a function to get the profile with a given uid, the ultimate goal of this update is to be able to put the list of all profiles private and only access some of the profiles
- DisplayLessonDetails.kt => add a component to display the lesson details, we will be able to reuse it when a tutor need to validate a requested lesson
- DisplayLesson.kt => add onClick to be able to add the navigation to TutorResponseLesson screen from the RequestedLesson screen
- LessonEditor.kt => fix the state of the lesson at the creation of it
- RequestedLesson.kt => add navigation to rge TutorResponseLesson screen when click on a lesson is triggered
- TutorLessonresponse.kt => the main new screen implemented in this PR
- NavigationsActions.kt => add a screen value for the TutorResponseLesson screen
- Overview.kt => update the category of lessons that are displayed with the updated lesson states
- OverViewTest.kt => fix the status of the lessons used for testing because pending lessons are not displayed for tutor (because pending means that there is no tutor assigned)

### How to test this PR
The tests for this screen are not yet available but you can already run the MainActivity with a Tutor profile and go to the "Find a student" section and select a Lesson to see the screen and interact with it.

### Screen example
![image](https://github.com/user-attachments/assets/215bbd4b-0736-43ca-9e57-b6f63b4f1dbc)
